### DISCUSSION TITLE
Is bevy_renet usable for p2p?

### DISCUSSION THREAD

**Discussion by tbillington** [lucaspoffo/renet#105] [Q&A] [answered]
Hey! just wondering whether bevy_renet is intended to support p2p connections.

I am experimenting with it for my game and am adding both client & server plugins so players can act as client or server.
Do you foresee any issues with doing this?

```rust
        group = group
            .add(RenetClientPlugin)
            .add(RenetServerPlugin)
            .add(NetcodeClientPlugin)
            .add(NetcodeServerPlugin);
```

**Comment by lucaspoffo** [comment]
It's possible to use renet with p2p, not intended but works. This is how the renet_steam works:
1 peer will act as the server and this peer will connect with all clients, the other peers will connect with the server-peer, so there is not really a benefit for using p2p, since clients will not send messages between themselves.

**Reply by lucaspoffo** [reply]
BTW, only one server can exist. Having all peers have the server plugin will not work, one peer needs to host the server

**Reply by tbillington** [reply]
Yep I would likely use renet_steam.

To my knowledge there's no way to remove a plugin, so if a player hosts a game then wants to join one after, would that cause an issue since the server plugin will have been added?

**Reply by lucaspoffo** [reply, marked as answer]
You can add both plugins without a problem, it runs only if the correct resources exist, so if there is no RenetServer, it will not run.
